File: stack&queue/programmers_42584/solution.py
def solution(prices):
    price_length = len(prices)
    answer = [0] * price_length
    stack = [] 

    for i in range(price_length):
        while stack and prices[stack[-1]] > prices[i]:
            top_index = stack.pop()
            answer[top_index] = i - top_index
        
        stack.append(i)

    while stack:
        top_index = stack.pop()
        answer[top_index] = (price_length - 1) - top_index
            
    return answer

File: stack&queue/programmers_42584/solution.js
function solution(prices) {
    const priceLength = prices.length;
    const answer = Array(priceLength).fill(0); 
    const stack = []; 

    for (let i = 0; i < priceLength; i++) {
        while (stack.length && prices[stack[stack.length - 1]] > prices[i]) {
            const topIndex = stack.pop(); 
            answer[topIndex] = i - topIndex; 
        }
        
        stack.push(i);
    }

    while (stack.length) {
        const topIndex = stack.pop();
        answer[topIndex] = (priceLength - 1) - topIndex; 
    }

    return answer;
}